status-resolver/src/main/java/com/ackee/versioupdatehandler/VersionStatusResolver.kt

```diff
@@ -0,0 +1,70 @@
+package com.ackee.versioupdatehandler
+
+import android.util.Log
+import androidx.fragment.app.FragmentManager
+import com.ackee.versioupdatehandler.model.DialogSettings
+import com.ackee.versioupdatehandler.model.VersionStatus
+import com.ackee.versioupdatehandler.model.VersionsConfiguration
+
+/**
+ * Class that contains the version comparing and update handling logic.
+ */
+class VersionStatusResolver(private val fetcher: VersionFetcher) {
+
+    suspend fun checkVersionStatus(version: Int): VersionStatus {
+        val versionConfig = fetcher.fetch()
+        return resolveStatus(version, versionConfig)
+    }
+
+    private fun resolveStatus(version: Int, configuration: VersionsConfiguration): VersionStatus {
+        var status = VersionStatus.UP_TO_DATE
+        if (version < configuration.minimalVersion()) {
+            status = VersionStatus.UPDATE_REQUIRED
+        } else if (version < configuration.currentVersion()) {
+            status = VersionStatus.UPDATE_AVAILABLE
+        }
+        Log.d("VersionStatusResolver", "Resolving version status... App version:" + version +
+            " minimal version: " + configuration.minimalVersion() +
+            " current version: " + configuration.minimalVersion() + " Status resolved:  " + status)
+        return status
+    }
+    /**
+     * Check for version and show dialog with customized visual settings via [DialogSettings]
```

Comment:
😐  there is duplicated javadoc and I don't know if we need any javadoc at all

Reply:
5ff6590de61b394853088024f80a06efab417818